plugins/calibrecompanion.koplugin/sha1.lua

```diff
@@ -0,0 +1,330 @@
+-- $Revision: 1.5 $
```

Comment:
Should we put this in base?

Reply:
Yep, could be used one day by other code than this plugin.

Reply:
Sounds cool. Could you please do it for me?. I'm not sure where to place the file and have no idea how to exclude it from shellcheck. It is spawning a lot of warnings

Reply:
I'd just put it in base/ffi. The bit manipulations are close enough, so to speak. Pinging @NiLuJe for another opinion.

You can exclude files in .luacheckrc, see https://github.com/koreader/koreader-base/blob/4a91eefe3a42f926ebe262658c258c9929544836/.luacheckrc#L12-L16

Reply:
Or with specific files rather than wildcards:

https://github.com/koreader/koreader/blob/0fb9113078bb727134a010178a1caedd79e47f8e/.luacheckrc#L134-L141

We could also say this is more like [luxl](https://github.com/koreader/koreader/blob/0fb9113078bb727134a010178a1caedd79e47f8e/frontend/luxl.lua)… although there I'm inclined to wonder why it's in front.

Reply:
That'd be a +1 on moving it to base, and looking into https://github.com/mpeterv/sha1 instead, which'll likely be much faster ;).

Reply:
Wow. Indeed your suggestion is several orders of magnitude faster. It isn't very noticeable on the usage here but in a `for i = 1, 1000` loop here are the results:

the one in this post:

```
real	0m13,375s
user	0m13,369s
sys	0m0,004s
```

mpeterv one using LuaJIT bitops:

```
real	0m0,030s
user	0m0,029s
sys	0m0,001s
```